### ISSUE TITLE
"Medium raw" methods

### ISSUE THREAD

**Issue by Bluenix2** [Rapptz/discord.py#6165]
## The Problem

With Discord rolling out intents, and limiting the data that a bot receives, it means that caches don't get filled. This complicates a lot of actions with using the library.

## Summary

I feel like the library should be able to provide more lower level, and "raw" methods, perhaps somewhat hidden though.

Currently, you can of course use the `Client.http` methods, but these are a bit _too_ low level. 

As an example, what I am suggesting is essentially moving the logic inside `Messageable.send` (that turns embeds into dicts and handles allowed mentions) into something like `Client.raw.send_message`.
The call stack would look like this: `Messageable.send` -> `Client.raw.send_message` -> `Client.http.send_message`.

`Client.raw.send_message` would take in IDs and the objects ( as opposed to `Client.http.send_message` which takes in dicts and makes the request ). This would change `Messageable.send` to only forward to `Client.raw.send_message`.

Again, like said. I am not so sure about the naming at all. Nor how it could be sustainably done (as in if it's good to have such `Client.raw`).
## Tl;Dr

With intents limiting discord.py's cache, there is a need for more lower level functions, but `Client.http` is too "raw".

**Comment by Rapptz** [comment]
If you're not using the guilds intent then this library won't be of much use for you in general. At least this is regarding the current messageable example.

With the new PartialMessage implementation I'm unsure what other "raw" things are desirable. As it stands right now the scope for this one is basically "use the undocumented http client" because anything else seems like it's been covered through the various methods provided in the library.

I have no direct plans to make the http client higher level. Models require a parent-child relationship with other models which makes returning rich objects from just their response JSON impractical and too much work. Which means this approach doesn't work well with any of the models. 

Take for example sending a message. This requires a rich Channel parameter to fill for Message.channel and is the way Message.guild gets filled as well. Then there's Message.author which is only a Member if it gets the upgrade from Message.guild, otherwise it's a regular old User. Through the gateway this upgrade is possible without, but using the raw HTTP API this is unexposed.

Ultimately this idea would require an entirely new modeling layer that would potentially harm the usability of those using the rich interface. There could be two different models, such as HTTPMessage vs Message but at this point duplicating literally all the code is way too much of an undertaking for what I perceive as little benefit. Discord.NET for example has a heavily polymorphic API and I do not enjoy it.
